dlib/test/dnn.cpp

```diff
@@ -2486,6 +2486,32 @@ namespace
         }
     }
 
+    void test_serialization()
+    {
+        print_spinner();
+
+        using net_type = loss_mean_squared<fc<1, input<matrix<double>>>>;
+        net_type net, net2;
+
+#if 0
+        // At least on MSVC, this fails with compiler error:
+        // "C2665: 'dlib::serialize': none of the 46 overloads could convert all the argument types"
+        {
+            std::ostringstream out;
+            dlib::serialize(net, out);
```

Comment:
It seems that the gcc on Travis does not compile this line either:
https://travis-ci.org/davisking/dlib/jobs/251527618#L639